### PR TITLE
Pass contract declarations to compile_test_prepared_db / Cairo update

### LINKED ISSUE #1629: Updates to compile_test_prepared_db api

### PR DESCRIPTION
**Pull request by maciektr**
- **Update Cairo**
- **Pass contract declarations to compile_test_prepared_db**

fixes #1629

### CODE REVIEW

File: scarb/src/compiler/compilers/test.rs [hunk 1]
```diff
@@ -58,14 +72,11 @@ impl Compiler for TestCompiler {
                 add_statements_code_locations: unit
                     .compiler_config
                     .unstable_add_statements_code_locations_debug_info,
+                contract_crate_ids: starknet.then_some(all_crate_ids),
+                executable_crate_ids: None,
+                contract_declarations: starknet.then_some(contracts.clone()),
```
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply scarb/src/compiler/compilers/test.rs:77]: Ok, upon further inspection, I see that `contract_declarations` and `contract_crate_ids` are required to be `None` if `starknet=false`.

I wonder though if the following statements still hold (I don't think they should) 🤔 
- `/// If not, all contracts from `contract_crate_ids` will be compiled.`
- `/// If not defined, all crates will be searched.`

File: scarb/src/compiler/compilers/test.rs [hunk 2]
```diff
@@ -58,14 +72,11 @@ impl Compiler for TestCompiler {
                 add_statements_code_locations: unit
                     .compiler_config
                     .unstable_add_statements_code_locations_debug_info,
+                contract_crate_ids: starknet.then_some(all_crate_ids),
```
[the remark this answers is not in the document]
Comment (maciektr) [reply scarb/src/compiler/compilers/test.rs:75]: Like above, you might want to instruct the compiler to either search in all crates, or to not search in any crates (regardless if starknet is enabled or not).

[the remark this answers is not in the document]
Comment (DelevoXDG) [reply scarb/src/compiler/compilers/test.rs:75]: I'm just not sure I understand why would we want to instruct compiler to search for contracts in all crates, if no contracts will be compiled anyway due to `starknet = false`

File: scarb/src/compiler/compilers/test.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (DelevoXDG) [reply scarb/src/compiler/compilers/test.rs:77]: Ok, that makes sense, just seemed to me that comments imply something else 😄, as in:

If `starknet=true`, `contract_declarations=None`, `contract_crate_ids=None` -> contracts will be searched in all crates, all contracts will be compiled

File: scarb/src/compiler/compilers/test.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (maciektr) [reply scarb/src/compiler/compilers/test.rs:75]: If starknet=false, those flags are ignored - if this clears things up.

**Review by DelevoXDG — APPROVED** [review]

**Pull request merged**